File: lib/MESCIUS/ComponentOne/WinForms/C1TouchToolKit/PolicySourceCodes/VisualBasicPowerPacksZoomPolicy.vb
Imports Microsoft.VisualBasic.PowerPacks
Imports System.Collections.Generic
Imports System.Text
Imports System.Windows.Forms
Imports GrapeCity.Win.MultiTouch

Public Class ShapeContainerZoomPolicy
	Inherits ZoomPolicy
	Public Overrides ReadOnly Property TargetType() As Type
		Get
			Return GetType(ShapeContainer)
		End Get
	End Property

	Public Overrides Sub ZoomBounds(control As Control, infos As ZoomBoundsInfo)
		Dim shapeContainer As ShapeContainer = TryCast(control, ShapeContainer)
		For Each shape As Shape In shapeContainer.Shapes
			If TypeOf shape Is SimpleShape Then
				Dim simpleShape As SimpleShape = TryCast(shape, SimpleShape)
				simpleShape.Bounds = infos.Zoom(simpleShape.Bounds)
			ElseIf TypeOf shape Is LineShape Then
				Dim lineShape As LineShape = TryCast(shape, LineShape)
				lineShape.StartPoint = infos.Zoom(lineShape.StartPoint)
				lineShape.EndPoint = infos.Zoom(lineShape.EndPoint)
			End If
		Next
		MyBase.ZoomBounds(control, infos)
	End Sub
End Class

File: lib/MESCIUS/ComponentOne/WinForms/C1TouchToolKit/PolicySourceCodes/VisualBasicPowerPacksZoomPolicy.cs
using Microsoft.VisualBasic.PowerPacks;
using System;
using System.Collections.Generic;
using System.Text;
using System.Windows.Forms;
using GrapeCity.Win.MultiTouch;

namespace GrapeCity.Win.MultiTouch.ZoomPolicys
{
    public class ShapeContainerZoomPolicy : ZoomPolicy
    {
        public override Type TargetType
        {
            get { return typeof(ShapeContainer); }
        }

        public override void ZoomBounds(Control control, ZoomBoundsInfo infos)
        {
            ShapeContainer shapeContainer = control as ShapeContainer;
            foreach (Shape shape in shapeContainer.Shapes)
            {
                if (shape is SimpleShape)
                {
                    SimpleShape simpleShape = shape as SimpleShape;
                    simpleShape.Bounds = infos.Zoom(simpleShape.Bounds);
                }
                else if (shape is LineShape)
                {
                    LineShape lineShape = shape as LineShape;
                    lineShape.StartPoint = infos.Zoom(lineShape.StartPoint);
                    lineShape.EndPoint = infos.Zoom(lineShape.EndPoint);
                }
            }
            base.ZoomBounds(control, infos);
        }
    }
}
